bufferserver/src/main/java/com/datatorrent/bufferserver/internal/DataList.java

```diff
@@ -79,27 +93,31 @@ public void rewind(int baseSeconds, int windowId) throws IOException
     for (DataListIterator dli : iterators.values()) {
       dli.rewind(processingOffset);
```

Comment:
dli.rewind() seems to assume that the processingOffset is the offset in the current DataListIterator buffer block (da.data), but this is not necessary true for the call. I am not sure if this is a bug or I am missing something here.